### PR TITLE
Fixes for a few FTL bugs

### PR DESCRIPTION
**Pull request by mathjazz**
Details in commit messages.

@jotes r?

### CODE REVIEW

File: pontoon/base/static/js/fluent_interface.js [hunk 1]
```diff
@@ -317,7 +317,7 @@ var Pontoon = (function (my) {
           } else {
             var attributes = ast.attributes;
             if (attributes && attributes.length) {
-              response = attributes[0].value.elements[0].value;
+              response = this.serializePlaceables(attributes[0].value.elements);
```
Comment (mathjazz) [inline pontoon/base/static/js/fluent_interface.js]: This fixes this bug:
https://pontoon.mozilla.org/sl/firefox-screenshots/server.ftl/?string=168505

Where the simplified version of the string is cut off if it contains placeables.

File: pontoon/base/static/js/translate.js [hunk 2]
```diff
@@ -1926,12 +1936,22 @@ var Pontoon = (function (my) {
           return;
         }
 
-        var source = $(this).find('.translation-clipboard').text();
+        // FTL Editor
+        if ($('#ftl').is('.active')) {
```
Comment (jotes) [inline pontoon/base/static/js/translate.js]: I think that at some point we'll have to split implementation of editors into separate classes. That's not a blocker, just an observation.

File: pontoon/base/static/js/translate.js [hunk 3]
```diff
@@ -1926,12 +1936,22 @@ var Pontoon = (function (my) {
...
 
-        $('#translation').val(source).focus();
-        self.moveCursorToBeginning();
-        self.updateCurrentTranslationLength();
-        self.updateInPlaceTranslation();
+        // Standard Editor
+        } else {
+          var source = $(this).find('.translation-clipboard').text();
+
+          $('#translation').val(source).focus();
```
Comment (jotes) [inline pontoon/base/static/js/translate.js]: I see that this code pattern occurs few times, maybe it's a good idea to create a method for that kind of render?

File: pontoon/base/static/js/translate.js [hunk 4]
```diff
@@ -1954,45 +1974,53 @@ var Pontoon = (function (my) {
...
+          button.prop('title', 'Approve');
+          button.parents('li.translated').removeClass('translated').addClass('suggested');
+          button.parents('li').find('.info a').prop('title', self.getApproveButtonTitle({
+            approved: false,
+            unapproved_user: self.user.display_name
+          }));
+
+          self.endLoader('Translation has been unapproved.');
+        }, function() {
+          self.endLoader("Couldn't unapprove this translation.");
```
Comment (jotes) [inline pontoon/base/static/js/translate.js]: I wonder if we should inform a user what was a reason of that error (e.g. internal server error vs. you don't have permissions etc.).

File: pontoon/base/static/js/translate.js [hunk 5]
```diff
@@ -1954,45 +1974,53 @@ var Pontoon = (function (my) {
...
+          }
+
+          if (entity.body && pf === 0) {
+            self.postMessage("SAVE", {
+              translation: data.translation.string,
+              id: entity.id
+            });
+          }
+
+          button.removeClass('unapprove').addClass('approve');
```
Comment (jotes) [inline pontoon/base/static/js/translate.js]: I hope that React/Angular will free us from that bloat of changing of state of elements inside the js code.

File: pontoon/base/static/js/translate.js (hunk 2 above)
Comment (mathjazz) [inline pontoon/base/static/js/translate.js]: Yessss!

File: pontoon/base/static/js/translate.js (hunk 4 above)
Comment (mathjazz) [inline pontoon/base/static/js/translate.js]: I agree, but I'm not sure if fits into this PR.

File: pontoon/base/static/js/translate.js (hunk 3 above)
Comment (mathjazz) [inline pontoon/base/static/js/translate.js]: Good point, will update!

**Pull request merged**